clang 10.0 compiler error

I get a compile error with clang 10.0:

```
meta/meta_fwd.hpp:286:20: error: expected concept name with optional arguments
[build]         { T{} } -> typename T::value_type;
[build]                    ^
```
Whats the fix?

--- a/include/meta/meta_fwd.hpp
+++ b/include/meta/meta_fwd.hpp
@@ -283,7 +283,7 @@ namespace meta
         typename detail::require_constant<T{}()>;
         requires T{}() == T::value;
 
-        { T{} } -> typename T::value_type;
+        { T{} } -> std::convertible_to<typename T::value_type>;
     };
     // clang-format on
 #endif // META_CONCEPT

